test/response_test.rb

```diff
@@ -852,9 +852,237 @@ class RubySamlTest < Minitest::Test
         signed_response = OneLogin::RubySaml::Response.new(document.to_s)
         settings.idp_cert = ruby_saml_cert_text
         signed_response.settings = settings
-        time = Time.parse("2015-03-18T04:50:24Z")
-        Time.stubs(:now).returns(time)
-        assert signed_response.is_valid?
+        Timecop.freeze(Time.parse("2015-03-18T04:50:24Z")) do
+          assert signed_response.is_valid?
+        end
+        assert_empty signed_response.errors
+      end
+    end
+
+    describe "retrieve nameID" do
+      it 'able when nameID inside the assertion' do
+        response_valid_signed.settings = settings
+        assert_equal "test@onelogin.com", response_valid_signed.name_id
+      end
+
+      it 'unable when encryptID inside the assertion but no private key' do
+          response_encrypted_nameid.settings = settings
+          assert_raises(OneLogin::RubySaml::ValidationError, "An EncryptedID found and no SP private key found on the settings to decrypt it") do
+            assert_equal "test@onelogin.com", response_encrypted_nameid.name_id    
+          end
+      end
+
+      it 'able when encryptID inside the assertion and settings has the private key' do
+        settings.private_key = ruby_saml_key_text
+        response_encrypted_nameid.settings = settings
+        assert_equal "test@onelogin.com", response_encrypted_nameid.name_id
+      end
+
+    end
+
+  end
+
+  describe '#is_assertion_encrypted?' do
+    it 'return false if none encrypted assertion is found' do
+      settings.certificate = ruby_saml_cert_text
+      settings.private_key = ruby_saml_key_text
+      response = OneLogin::RubySaml::Response.new(response_document_valid_signed, :settings => settings)
+      assert !response.send(:assertion_encrypted?)
```

Comment:
then you don't need theses specs. The others will let you know if this method fails. The problem writing specs for private methods is that these kind of methods were made to be changed internally, so it doesn't make sense to write them.